### PR TITLE
fix: refuse to start with default secret on non debug envs

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Impose harsher measures when a default SECRET_KEY is identified.
On non development environments on Flask debug mode Superset will refuse to start

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — APPROVED** [review]
LGTM

**Review by craig-rueda — APPROVED** [review]

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ assists people when migrating to a new version.
 
 ## Next
 
+- [23186](https://github.com/apache/superset/pull/23186): Superset will refuse to start if a default `SECRET_KEY` is detected on a non Flask debug setting.
```
Comment (eschutho) [inline UPDATING.md:27]: This looks like a potentially breaking change. If we want to put this in a patch or minor release, can we issue a warning first?

File: superset/initialization/__init__.py [hunk 2]
```diff
@@ -471,6 +472,20 @@ def check_secret_key(self) -> None:
...
+                or self.superset_app.config["TESTING"]
+                # There must be a better way
+                or "pytest" in sys.modules
+            ):
+                logger.warning("Debug mode identified with default secret key")
+                log_default_secret_key_warning()
+                return
+            log_default_secret_key_warning()
+            logger.error("Refusing to start due to insecure SECRET_KEY")
+            sys.exit(1)
```
Comment (eschutho) [inline superset/initialization/__init__.py:487]: maybe we can just take out the exit for the patch?

File: UPDATING.md (hunk 1 above)
Comment (dpgaspar) [reply UPDATING.md:27]: a warning to the community? I think so, changing your default secret_key is really mandatory

File: superset/initialization/__init__.py (hunk 2 above)
Comment (dpgaspar) [reply superset/initialization/__init__.py:487]: we can't unfortunately

**Review by eschutho — APPROVED** [review]

**Pull request merged**